orc8r/cloud/go/obsidian/swagger/combine_test.go

```diff
@@ -0,0 +1,129 @@
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package swagger_test
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"magma/orc8r/cloud/go/obsidian/swagger"
+	"magma/orc8r/cloud/go/obsidian/swagger/protos"
+	"magma/orc8r/cloud/go/orc8r"
+	swagger_lib "magma/orc8r/cloud/go/swagger"
+	"magma/orc8r/cloud/go/test_utils"
+	"magma/orc8r/lib/go/registry"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v2"
+)
+
+func Test_GetCommonSpec(t *testing.T) {
+	specPath := "/etc/magma/configs/orc8r/swagger_specs"
+	commonSpecDir := "/etc/magma/configs/orc8r/swagger_specs/common"
+	commonSpecFilePath := "/etc/magma/configs/orc8r/swagger_specs/common/swagger-common.yml"
+
+	os.RemoveAll(specPath)
+	defer os.RemoveAll(specPath)
+
+	err := os.MkdirAll(commonSpecDir, os.ModePerm)
+	assert.NoError(t, err)
+
+	commonSpec := swagger_lib.Spec{
+		Tags: []swagger_lib.TagDefinition{{Name: "Tag Common"}},
+	}
+	yamlCommon := marshalToYAML(t, commonSpec)
+
+	err = ioutil.WriteFile(commonSpecFilePath, []byte(yamlCommon), 0644)
+	assert.NoError(t, err)
+
+	actual, err := swagger.GetCommonSpec()
+	assert.NoError(t, err)
+	assert.Equal(t, yamlCommon, actual)
+}
+
+func Test_GetCombinedSwaggerSpecs(t *testing.T) {
+	commonTag := swagger_lib.TagDefinition{Name: "Tag Common"}
+	commonSpec := swagger_lib.Spec{Tags: []swagger_lib.TagDefinition{commonTag}}
+	yamlCommon := marshalToYAML(t, commonSpec)
+
+	// Success with no registered servicers
+	expectedSpec := swagger_lib.Spec{
+		Tags: []swagger_lib.TagDefinition{commonTag},
+	}
+	expectedYaml := marshalToYAML(t, expectedSpec)
+
+	combined, err := swagger.GetCombinedSpec(yamlCommon)
+	assert.NoError(t, err)
+
+	assert.Equal(t, expectedYaml, combined)
+
+	// Success with registered servicers
+	tags := []swagger_lib.TagDefinition{
+		{Name: "Tag 1"},
+		{Name: "Tag 2"},
+		{Name: "Tag 3"},
+	}
+	services := []string{"test_spec_service1", "test_spec_service2", "test_spec_service3"}
+
+	expectedSpec = swagger_lib.Spec{
+		Tags: []swagger_lib.TagDefinition{tags[0], tags[1], tags[2], commonTag},
+	}
+	expectedYaml = marshalToYAML(t, expectedSpec)
+
+	for i, s := range services {
+		registerTestSpecServicer(t, s, tags[i])
+	}
+
+	combined, err = swagger.GetCombinedSpec(yamlCommon)
+	assert.NoError(t, err)
+
+	assert.Equal(t, expectedYaml, combined)
+
+	// Success even with merge warnings(duplicate tag)
+	serviceDuplicate := "test_spec_service_dup"
+	registerTestSpecServicer(t, serviceDuplicate, tags[0])
+
+	combined, err = swagger.GetCombinedSpec(yamlCommon)
+	assert.NoError(t, err)
+
+	assert.Equal(t, expectedYaml, combined)
+
+	// Clean up registry
+	registry.RemoveService(services[0])
+	registry.RemoveService(services[1])
+	registry.RemoveService(services[2])
+	registry.RemoveService(serviceDuplicate)
```

Comment:
Move these lines to directly before you register the first test servicer, and do `defer registry.RemoveServicesWithLabel(orc8r.SpecServicerLabel)`